Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -342,44 +342,66 @@ namespace AZ::DocumentPropertyEditor
+                auto containerAttribute = attributes.Find(AZ::Reflection::DescriptorAttributes::Container);
+                if (!containerAttribute.IsNull())
                 {
-                    m_builder.BeginPropertyEditor<Nodes::ContainerActionButton>();
-                    m_builder.Attribute(Nodes::ContainerActionButton::Action, Nodes::ContainerAction::AddElement);
-                    m_builder.AddMessageHandler(m_adapter, Nodes::ContainerActionButton::OnActivate.GetName());
-                    m_builder.EndPropertyEditor();
+                    auto container = AZ::Dom::Utils::ValueToTypeUnsafe<AZ::SerializeContext::IDataContainer*>(containerAttribute);
+                    m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer{ container, access.Get() });
+                    size_t containerSize = container->Size(access.Get());
+                    if (containerSize == 1)
+                    {
+                        m_builder.Label(AZStd::string::format("%s (1 element)", labelAttribute.data()));
+                    }
+                    else
+                    {
+                        m_builder.Label(AZStd::string::format("%s (%zu elements)", labelAttribute.data(), container->Size(access.Get())));
+                    }
 
-                    m_builder.BeginPropertyEditor<Nodes::ContainerActionButton>();
-                    m_builder.Attribute(Nodes::ContainerActionButton::Action, Nodes::ContainerAction::Clear);
-                    m_builder.AddMessageHandler(m_adapter, Nodes::ContainerActionButton::OnActivate.GetName());
-                    m_builder.EndPropertyEditor();
+                    if (!container->IsFixedSize())
+                    {
+                        m_builder.BeginPropertyEditor<Nodes::ContainerActionButton>();
+                        m_builder.Attribute(Nodes::ContainerActionButton::Action, Nodes::ContainerAction::AddElement);
+                        m_builder.AddMessageHandler(m_adapter, Nodes::ContainerActionButton::OnActivate.GetName());
+                        m_builder.EndPropertyEditor();
+
+                        m_builder.BeginPropertyEditor<Nodes::ContainerActionButton>();
+                        m_builder.Attribute(Nodes::ContainerActionButton::Action, Nodes::ContainerAction::Clear);
+                        m_builder.AddMessageHandler(m_adapter, Nodes::ContainerActionButton::OnActivate.GetName());
+                        m_builder.EndPropertyEditor();
+                    }
                 }
-            }
-
-            AZ::Dom::Value instancePointerValue = AZ::Dom::Utils::MarshalTypedPointerToValue(access.Get(), access.GetType());
-            VisitValue(
-                instancePointerValue, access.Get(), attributes,
-                [](const Dom::Value& newValue)
+                else
                 {
-                    return newValue;
-                },
-                false);
+                    m_builder.Label(labelAttribute.data());
```

Comment:
@NicholasVanSickle This is the general non-string*, non-container case, I believe.

Reply:
Ok yeah, I think this is fine, +1ing